### PR TITLE
[4.0] WebAsset for layouts

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes
This replaces htmlhelper to webaset for layouts

Updated layouts:
```
joomla/html/batch/item
joomla/html/batch/language
joomla/html/batch/adminlanguage
joomla/form/field/calendar
joomla/form/field/color/slider
joomla/form/field/password
joomla/form/renderfield
joomla/toolbar/containeropen
joomla/searchtools/default/filters
joomla/edit/frontediting_modules
joomla/content/options_default
plugins/system/webauthn/manage
```

### Testing Instructions
Apply patch. Run `npm install`
Click around, check calendar, template style colors, batch,  fields "showon",
all should work as before

### Expected result
All work

### Actual result
All work

### Documentation Changes Required
nope

ref #22435

### CODE REVIEW

File: layouts/joomla/edit/frontediting_modules.php [hunk 1]
```diff
@@ -70,6 +70,8 @@
 	// Load stylesheet and javascript to head:
 	HTMLHelper::_('bootstrap.popover');
 
-	HTMLHelper::_('stylesheet', 'legacy/frontediting.css', array('version' => 'auto', 'relative' => true));
-	HTMLHelper::_('script', 'legacy/frontediting.min.js', array('version' => 'auto', 'relative' => true));
+	/** @var Joomla\CMS\WebAsset\WebAssetManager $wa */
+	$wa = Factory::getApplication()->getDocument()->getWebAssetManager();
+	$wa->useStyle('joomla.frontediting')
```
Comment (wilsonge) [inline layouts/joomla/edit/frontediting_modules.php]: would it make sense to make these two a preset?

Comment (Fedik) [reply layouts/joomla/edit/frontediting_modules.php]: good point, may have :) 
I will update

Comment (Fedik) [reply layouts/joomla/edit/frontediting_modules.php]: updated

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on ff7a94f7c90f0c528bd78816121fd7f6efba964c<br /><br />tested with the new patchtester 4 
so i'vent runned npm install<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/28431">issues.joomla.org/tracker/joomla-cms/28431</a>.</sub>

**Comment by Fedik** [comment]
I have added all "showon" cases to this PR, to make "showon" use WebAsset in all places,
this now fully cover #28456

**Comment by wilsonge** [comment]
LGTM! Thanks!

**Comment by bugsmafia** [comment]
This began to cause a drop wherever the password field is involved.

An error has occurred.
0 There is no "field.passwordview" asset of a "script" type in the registry.
iYUFsjZDTUmveTa7WhpYUQ 1

**Pull request merged**